File: app/src/main/java/com/listocalixto/android/ecommerce/models/Product.kt
package com.listocalixto.android.ecommerce.models

import com.google.gson.Gson
import com.google.gson.annotations.SerializedName

data class Product(
    @SerializedName("id") val id: String? = null,
    @SerializedName("name") val name: String,
    @SerializedName("description") val description: String,
    @SerializedName("price") val price: Double,
    @SerializedName("image00") val image00: String? = null,
    @SerializedName("image01") val image01: String? = null,
    @SerializedName("image02") val image02: String? = null,
    @SerializedName("id_category") val idCategory: String,
    @SerializedName("quantity") val quantity: Int? = null
) {

    fun toJson(): String {
        return Gson().toJson(this)
    }

    override fun toString(): String {
        return "Product(id='$id', name='$name', description='$description', price=$price, image00='$image00', image01='$image01', image02='$image02', idCategory='$idCategory', quantity=$quantity)"
    }


}

File: backend/models/product.js
const db = require('../config/config')

const Product = {};

Product.create = (product) => {
    const sql = `
    INSERT INTO
        products(
            name,
            description,
            price,
            image00,
            image01,
            image02,
            id_category,
            created_at,
            updated_at
        )
    VALUES($1, $2, $3, $4, $5, $6, $7, $8, $9) RETURNING id
    `;

    return db.oneOrNone(sql, [
        product.name,
        product.description,
        product.price,
        product.image00,
        product.image01,
        product.image02,
        product.id_category,
        new Date(),
        new Date()
    ]);

}

Product.update = (product) => {
    const sql = `
    UPDATE
        products
    SET
        name = $2,
        description = $3,
        price = $4,
        image00 = $5,
        image01 = $6,
        image02 = $7,
        id_category = $8,
        updated_at = $9
    WHERE
        id = $1
    `;

    return db.none(sql, [
        product.id,
        product.name,
        product.description,
        product.price,
        product.image00,
        product.image01,
        product.image02,
        product.id_category,
        new Date()
    ]);

}

module.exports = Product;